File: allinone-kotlin/app/src/main/java/com/example/allinone/data/Transaction.kt
package com.example.allinone.data

import java.util.Date

data class Transaction(
    val id: Long = 0,
    val amount: Double = 0.0,
    val type: String = "",
    val description: String = "",
    val isIncome: Boolean = false,
    val date: Date = Date(),
    val category: String = "",
    val relatedRegistrationId: Long? = null  // Reference to linked registration if applicable
) 

File: src/features/transactions/types/Transaction.ts
export type TransactionCurrency = 'TRY' | 'AED' | 'USD';

export const TRANSACTION_CURRENCIES: TransactionCurrency[] = ['TRY', 'AED', 'USD'];

export interface Transaction {
  id: string; // We'll keep as string for React Native compatibility
  amount: number;
  currency: TransactionCurrency;
  type: string; // Category name (same as category)
  description: string; // Not nullable like Kotlin
  isIncome: boolean;
  date: string; // ISO string for React Native
  category: string;
  relatedRegistrationId?: number; // Reference to linked registration if applicable
  relatedInvestmentId?: string; // Reference to linked investment if applicable
}
